extensions/email/src/main/java/org/mule/extension/email/internal/util/EmailUtils.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.util;
+
+import static javax.mail.Part.ATTACHMENT;
+import static org.mule.extension.email.internal.util.EmailConstants.MULTIPART;
+import static org.mule.extension.email.internal.util.EmailConstants.TEXT;
+import org.mule.extension.email.internal.EmailAttributes;
+import org.mule.extension.email.internal.exception.EmailException;
+import org.mule.runtime.api.message.MuleMessage;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.activation.DataHandler;
+import javax.mail.Address;
+import javax.mail.BodyPart;
+import javax.mail.Multipart;
+import javax.mail.Part;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+/**
+ * this class contains common methods for email handling.
+ *
+ * @since 4.0
+ */
+public class EmailUtils
+{
+
+    /**
+     * Converts a simple {@link String} representing an address into an {@link InternetAddress} instance
+     *
+     * @param address the string to be converted.
+     * @return a new {@link InternetAddress} instance.
+     */
+    public static Address toAddress(String address)
+    {
+        try
+        {
+            return new InternetAddress(address);
+        }
+        catch (AddressException e)
+        {
+            throw new EmailException(String.format("Error while creating %s InternetAddress", address));
+        }
+    }
+
+    /**
+     * Converts a {@link List} of {@link String}s representing email addresses into an {@link InternetAddress} array.
+     *
+     * @param addresses the list to be converted.
+     * @return a new {@link Address}[] instance.
+     */
+    public static Address[] toAddressArray(List<String> addresses)
+    {
+        return addresses.stream().map(EmailUtils::toAddress).toArray(Address[]::new);
```

Comment:
`.toArray(Address[]::new);` is equivalent to `toArray(length -> new Address[lenght])`